hoomd/pytest/test_simulation.py

```diff
@@ -0,0 +1,156 @@
+import hoomd
+import numpy as np
+import pytest
+from copy import deepcopy
+try:
+    import gsd.hoomd
+    skip_gsd = False
+except ImportError:
+    skip_gsd = True
+
+
+@pytest.fixture(scope="function")
+def get_snapshot(device):
+    def make_snapshot(n=10, particle_types=['A']):
+        s = hoomd.snapshot.Snapshot(device.comm)
+        if s.exists:
+            s.configuration.box = [20, 20, 20, 0, 0, 0]
+            s.particles.N = n
+            s.particles.position[:] = np.random.uniform(-10, 10, size=(n, 3))
+            s.particles.types = particle_types
+        return s
+    return make_snapshot
+
+
+def make_gsd_snapshot(hoomd_snapshot):
+    s = gsd.hoomd.Snapshot()
+    for attr in dir(hoomd_snapshot):
+        if attr[0] != '_' and attr not in ['exists', 'replicate']:
+            for prop in dir(getattr(hoomd_snapshot, attr)):
+                if prop[0] != '_':
+                    # s.attr.prop = hoomd_snapshot.attr.prop
+                    setattr(getattr(s, attr), prop,
+                            getattr(getattr(hoomd_snapshot, attr), prop))
+    return s
+
+
+def set_types(s, inds, particle_types, particle_type):
+    for i in inds:
+        s.particles.typeid[i] = particle_types.index(particle_type)
+
+
+def update_positions(snap):
+    if snap.exists:
+        noise = 0.01
+        rs = np.random.RandomState(0)
+        mean = [0] * 3
+        var = noise * noise
+        cov = np.diag([var, var, var])
+        shape = snap.particles.position.shape
+        snap.particles.position[:] += rs.multivariate_normal(mean, cov,
+                                                             size=shape[:-1])
+    return snap
+
+
+def assert_equivalent_snapshots(snap1, snap2):
+    for attr in dir(snap2):
+        if attr[0] != '_' and attr not in ['exists', 'replicate']:
+            for prop in dir(getattr(snap2, attr)):
+                if prop[0] != '_':
+                    if prop == 'types':
+                        assert getattr(getattr(snap1, attr), prop) == \
+                            getattr(getattr(snap2, attr), prop)
+                    else:
+                        np.testing.assert_allclose(getattr(getattr(snap1, attr),
+                                                           prop),
+                                                   getattr(getattr(snap2, attr),
+                                                           prop))
+
+
+def assert_equivalent_boxes(box1, box2):
```

Comment:
When the `feature/new-box-resize` gets merged we can just use `box1 == box2`. Unfortunately for now this is required though.